bundles/org.openhab.binding.hapero/src/main/java/org/openhab/binding/hapero/internal/handler/HaperoTankHandler.java

```diff
@@ -0,0 +1,135 @@
+    private static final int CHANNEL_TEMPERATUREBOTTOM_INDEX = 1;
+    private static final int CHANNEL_ONTEMPERATURE_INDEX = 2;
+    private static final int CHANNEL_OFFTEMPERATURE_INDEX = 3;
+    private static final int CHANNEL_PUMP_INDEX = 4;
+    private static final int CHANNEL_SWITCHVALVE_INDEX = 5;
+    private static final int CHANNEL_CHARGING_INDEX = 6;
+
+    /* Some temperature values are scaled down by 10 */
+    private static final int TEMPERATURE_VALUE_SCALE = 10;
+
+    /**
+     * Constructor
+     *
+     * @param thing
+     */
+    public HaperoTankHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    protected @Nullable State getStateForChannel(Channel channel, Device device) {
+        State state = null;
+        Float value;
+        Boolean valueBoolean;
+        String channelId = channel.getUID().getIdWithoutGroup();
+
+        switch (channelId) {
+            case HaperoBindingConstants.CHANNEL_TEMPERATURETOP:
+                value = device.getFloat(CHANNEL_TEMPERATURETOP_INDEX);
+                if (value != null) {
+                    value /= TEMPERATURE_VALUE_SCALE;
+                    state = new QuantityType<Temperature>(value, SIUnits.CELSIUS);
+                }
+                break;
+            case HaperoBindingConstants.CHANNEL_TEMPERATUREBOTTOM:
+                value = device.getFloat(CHANNEL_TEMPERATUREBOTTOM_INDEX);
+                if (value != null) {
+                    value /= TEMPERATURE_VALUE_SCALE;
+                    state = new QuantityType<Temperature>(value, SIUnits.CELSIUS);
+                }
+                break;
+            case HaperoBindingConstants.CHANNEL_ONTEMPERATURE:
+                value = device.getFloat(CHANNEL_ONTEMPERATURE_INDEX);
+                if (value != null) {
+                    value /= TEMPERATURE_VALUE_SCALE;
+                    state = new QuantityType<Temperature>(value, SIUnits.CELSIUS);
+                }
+                break;
+            case HaperoBindingConstants.CHANNEL_OFFTEMPERATURE:
+                value = device.getFloat(CHANNEL_OFFTEMPERATURE_INDEX);
+                if (value != null) {
+                    value /= TEMPERATURE_VALUE_SCALE;
+                    state = new QuantityType<Temperature>(value, SIUnits.CELSIUS);
+                }
+                break;
+            case HaperoBindingConstants.CHANNEL_PUMP:
+                valueBoolean = device.getBoolean(CHANNEL_PUMP_INDEX);
+                if (valueBoolean != null) {
+                    if (valueBoolean) {
+                        state = OnOffType.ON;
+                    } else {
+                        state = OnOffType.OFF;
+                    }
+                }
+                break;
+            case HaperoBindingConstants.CHANNEL_SWITCHVALVE:
+                valueBoolean = device.getBoolean(CHANNEL_SWITCHVALVE_INDEX);
+                if (valueBoolean != null) {
+                    if (valueBoolean) {
```

Comment:
`state = valueBoolean ? OpenClosedType.OPEN : OpenClosedType.CLOSED;`

Reply:
Ahm yes. Lets say it was late.